kubectl-plugin/pkg/cmd/create/create_cluster.go

```diff
@@ -0,0 +1,189 @@
+	}
+
+	err = cmd.MarkFlagRequired("worker-grp-name")
+	if err != nil {
+		log.Fatalf("Failed to mark worker-grp-name flag as required %v", err)
+	}
+	options.configFlags.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func (options *CreateClusterOptions) Complete(cmd *cobra.Command, args []string) error {
+	if *options.configFlags.Namespace == "" {
+		*options.configFlags.Namespace = "default"
+	}
+
+	if len(args) != 1 {
+		return cmdutil.UsageErrorf(cmd, "%s", cmd.Use)
+	}
+	options.clusterName = args[0]
+
+	return nil
+}
+
+func (options *CreateClusterOptions) Validate() error {
+	config, err := options.configFlags.ToRawKubeConfigLoader().RawConfig()
+	if err != nil {
+		return fmt.Errorf("Error retrieving raw config: %w", err)
+	}
+	if len(config.CurrentContext) == 0 {
+		return fmt.Errorf("no context is currently set, use %q to select a new one", "kubectl config use-context <context>")
+	}
+
+	return nil
+}
+
+func (options *CreateClusterOptions) Run(ctx context.Context, factory cmdutil.Factory) error {
+	var yamlFileToApply []byte
+
+	// Will generate yaml file
+	fmt.Printf("Generating Ray Cluster yaml...\n")
+	rayClusterObject := generation.RayClusterYamlObject{
+		Namespace:      *options.configFlags.Namespace,
+		ClusterName:    options.clusterName,
+		RayVersion:     options.rayVersion,
+		Image:          options.image,
+		HeadCPU:        options.headCPU,
+		HeadMemory:     options.headMemory,
+		WorkerGrpName:  options.workerGrpName,
+		WorkerReplicas: options.workerReplicas,
+		WorkerCPU:      options.workerCPU,
+		WorkerMemory:   options.workerMemory,
+	}
+
+	rayClusterYamlData, err := rayClusterObject.GenerateRayClusterYaml()
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Generated Yaml: \n")
+	fmt.Printf("%s\n", string(rayClusterYamlData))
+
+	// Wait for user to confirm
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Printf("Enter 'yes/y' to continue: \n")
+	input, err := reader.ReadString('\n')
+	if err != nil {
+		return err
+	}
+	input = strings.ToLower(strings.TrimSpace(input))
+	if input != "yes" && input != "y" {
+		return fmt.Errorf("Unknown input: %s", input)
+	}
+
+	// TODO: Decide whether to save yaml to file or not.
+
+	yamlFileToApply = rayClusterYamlData
+
+	decoder := yaml.NewDecodingSerializer(unstructured.UnstructuredJSONScheme)
+	rayClusterRuntimeObj := &unstructured.Unstructured{}
+	_, gvk, err := decoder.Decode(yamlFileToApply, nil, rayClusterRuntimeObj)
+	if err != nil {
+		return err
+	}
+
+	dynamicClient, err := factory.DynamicClient()
+	if err != nil {
+		return fmt.Errorf("dynamic client failed to initialize: %w", err)
+	}
+
+	// Applying the YAML
+	result, err := dynamicClient.Resource(gvk.GroupVersion().WithResource("rayclusters")).Namespace(*options.configFlags.Namespace).Create(ctx, rayClusterRuntimeObj, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("Failed to create Ray Cluster with: %w", err)
+	}
+	fmt.Printf("Created Ray Cluster: %s\n", result.GetName())
```

Comment:
Maybe for a follow-up, consider also polling for the RayCluster status to be ready and then printing additional commands on how to connect to the Ray cluster using `kubectl ray session`